src/rsz/src/Resizer.cc

```diff
@@ -375,6 +375,13 @@ bool Resizer::removeBuffer(Instance* buffer,
                "remove {}",
                db_network_->name(buffer));
     buffer_removed = true;
+
+    odb::dbNet* db_survivor = db_network_->staToDb(survivor);
+    odb::dbNet* db_removed = db_network_->staToDb(removed);
+    if (removed) {
+      db_survivor->mergeNet(db_removed);
```

Comment:
The db should do the merge and the sta merged related code here should become unnecessary (aside from parasitics).